rotkehlchen/exchanges/binance.py

```diff
@@ -522,30 +529,91 @@ def _deserialize_asset_movement(self, raw_data: Dict[str, Any]) -> Optional[Asse
 
         return None
 
+    def _api_query_dict_within_time_delta(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            time_delta: Timestamp,
+            method: Literal["depositHistory.html", "withdrawHistory.html"],
+    ) -> List[Dict[str, Any]]:
+        """Request via `api_query_dict()` from `start_ts` `end_ts` using a time
+        delta (offset) less than `time_delta`.
+
+        Be aware of:
+          - If `start_ts` equals zero, the Binance launch timestamp is used
+          (from BINANCE_LAUNCH_TS). This value is not stored in the
+          `used_query_ranges` table, but 0.
+          - Timestamps are converted to milliseconds.
+        """
+        if method == 'depositHistory.html':
+            query_schema = 'depositList'
+        elif method == 'withdrawHistory.html':
+            query_schema = 'withdrawList'
+        else:
+            raise AssertionError(f'Unexpected binance method case: {method}.')
+
+        results: List[Dict[str, Any]] = []
+        # Create required time references in milliseconds
+        offset = time_delta * 1000 - 1  # less than time_delta
+        if start_ts == Timestamp(0):
+            from_ts = BINANCE_LAUNCH_TS * 1000
+        else:
+            from_ts = start_ts * 1000
+
+        to_ts = (
```

Comment:
in the entire function and inside the loop you are doing `end_ts * 1000`. Avoid all these unnecessary extra multiplications by doing that once at the start. `end_ts = end_ts * 1000`

Reply:
Ah! If you don't mind to re-reference `end_ts` (from seconds to milliseconds) I will do it that way. I tried to get rid of the `_ms` variables but keeping the original `start_ts` and `end_ts`.

Reply:
Why would anyone mind. I do mind unnecessary operations more than anything.